### DISCUSSION TITLE
Time variable in ParsedFunction not working

### DISCUSSION THREAD

**Discussion by ashishdhole** [idaholab/moose#27708] [Q&A Modules: Phase field]
Hello. 
I am trying to change an aux variable with respect to the simulation time using a custom function defined by ParsedFunction. 
Following is the snippet of my code. 
```
[AuxVariables]
  [bnds]
    order = FIRST
    family = LAGRANGE
  []
  [c]
    [InitialCondition]
      type = FunctionIC
      function = time
    []
  []
[]
[Functions]
  [time]
    type = ParsedFunction
    expression = 't'
  []
[]
```
but I do not see variable 'c' changing with time. 
Following is the output (.csv).
```
time | c | gr1area
-- | -- | --
1 | 0 | 654375.0005
2 | 0 | 654375.001
3 | 0 | 654375.0014
4 | 0 | 654375.0019
.
.
.

95 | 0 | 654375.0456
96 | 0 | 654375.0461
97 | 0 | 654375.0466
98 | 0 | 654375.0471
99 | 0 | 654375.0475
100 | 0 | 654375.048
```
Best Regards,
Ashish

**Comment by GiudGiud** [comment]
Hello

Initial conditions are evaluated only right before the very first  time step

use an auxiliary kernel (AuxKernel) instead for this

**Comment by ashishdhole** [comment]
Hello Guillaume,

Thank you very much. Now it is working as expected. 

Best Regards
Ashish
